common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java

```diff
@@ -39,24 +39,62 @@
 
     @SuppressWarnings("UnusedCatchParameter")
     private static InternalLoggerFactory newDefaultFactory(String name) {
-        InternalLoggerFactory f;
+        InternalLoggerFactory f = useSlf4JLoggerFactory(name);
+        if (f != null) {
+            return f;
+        }
+
+        f = useLog4J2LoggerFactory(name);
+        if (f != null) {
+            return f;
+        }
+
+        f = useLog4JLoggerFactory(name);
+        if (f != null) {
+            return f;
+        }
+
+        return useJdkLoggerFactory(name);
+    }
+
+    private static InternalLoggerFactory useSlf4JLoggerFactory(String name) {
         try {
-            f = new Slf4JLoggerFactory(true);
+            InternalLoggerFactory f = new Slf4JLoggerFactory(true);
             f.newInstance(name).debug("Using SLF4J as the default logging framework");
-        } catch (Throwable ignore1) {
-            try {
-                f = Log4J2LoggerFactory.INSTANCE;
-                f.newInstance(name).debug("Using Log4J2 as the default logging framework");
-            } catch (Throwable ignore2) {
-                try {
-                    f = Log4JLoggerFactory.INSTANCE;
-                    f.newInstance(name).debug("Using Log4J as the default logging framework");
-                } catch (Throwable ignore3) {
-                    f = JdkLoggerFactory.INSTANCE;
-                    f.newInstance(name).debug("Using java.util.logging as the default logging framework");
-                }
-            }
+            return f;
+        } catch (LinkageError ignore) {
+            return null;
+        } catch (Exception ignore) {
```

Comment:
This can catch the more specific `(ReflectiveOperationException | SecurityException ignore)` unless for some reason `Slf4JLoggerFactory` is throwing java.lang.Exception.

Reply:
@elharo I did decide to use `Exception` as `ReflectiveOperationeException` not exists in JDK6 :/

Reply:
Netty still supports JDK 6? Impressive. In that case, please add a comment to that effect here for anyone who reads this code in the future. And then LGTM.

Reply:
yes ... unfortunally we do :/